$NetBSD: patch-ai,v 1.2 2015/12/29 23:34:53 dholland Exp $

Get pointer-sized ints portably.
XXX: why HAVE_INTTYPES_H vs. include <stdint.h>?

--- src/jstypes.h.orig	2006-07-24 15:21:49.000000000 -0400
+++ src/jstypes.h
@@ -387,8 +387,14 @@ typedef JSUint8 JSPackedBool;
 /*
 ** A JSWord is an integer that is the same size as a void*
 */
+#ifdef HAVE_INTTYPES_H
+#include <stdint.h>
+typedef intptr_t JSWord;
+typedef uintptr_t JSUword;
+#else
 typedef long JSWord;
 typedef unsigned long JSUword;
+#endif
 
 #include "jsotypes.h"
 
